actionview/lib/action_view/helpers/date_helper.rb

```diff
@@ -90,12 +90,17 @@ module DateHelper
       #   distance_of_time_in_words(from_time, from_time + 50.minutes, scope: 'datetime.distance_in_words.short') # => "an hour"
       #   distance_of_time_in_words(from_time, from_time + 3.hours, scope: 'datetime.distance_in_words.short')    # => "3 hours"
       def distance_of_time_in_words(from_time, to_time = 0, options = {})
+        if [from_time, to_time].any?(&:nil?)
+          raise ArgumentError, "given time must not be nil"
+        end
+
         options = {
           scope: :'datetime.distance_in_words'
         }.merge!(options)
 
         from_time = from_time.to_time if from_time.respond_to?(:to_time)
         to_time = to_time.to_time if to_time.respond_to?(:to_time)
+        from_time, to_time = Time.at(from_time), Time.at(to_time)
```

Comment:
Should be fine - the two lines above convert any `AS::TimeWithZone` instances to local times so `Time.at` just essentially copies the values to a new `Time` instance. If there was a problem it's probably an underlying bug in `to_time`.

Thinking about it, maybe it might make things simpler if we define `Numeric#to_time` ? e.g:

``` ruby
class Numeric
  def to_time
    Time.at(self)
  end
end
```

This way we can eliminate the two `Time.at` calls, add a check for whether the two arguments respond to `to_time` and raise if they don't, eliminating the nil check.